include/utils/kahan.hpp

```diff
@@ -43,4 +45,94 @@ T kahan(InputIt first, InputIt last, T init) {
     return sum;
 }
 
+template<class InputIt, class T>
+T kahanBabushka(InputIt first, InputIt last, T init) {
+    T sum = init;
+    T cs = 0.;
+    T ccs = 0.;
+    T c = 0.;
+    T cc = 0.;
+
+    for (auto input = first; input != last; input++) {
+        T t = sum + *input;
+        if(std::abs(sum) >= std::abs(*input))
+            c = (sum - t) + *input;
+        else
+            c = (*input - t) + sum;
+        sum = t;
+        t = cs + c;
+        if(std::abs(cs) >= std::abs(c))
+            cc = (cs - t) + c;
+        else
+            cc = (c - t) + cs;
+        cs = t;
+        ccs = ccs + cc;
+    }
+    return sum + cs + ccs;
+}
+
+std::tuple<double,double,int,int> kahanBabushkaTuple(std::vector<BankedParticle>::iterator first, std::vector<BankedParticle>::iterator last) {
+
+    double sum1 = 0.;
+    double sum2 = 0.;
+
+    int sum3 = 0;
+    int sum4 = 0;
+
+    double cs1 = 0.;
+    double ccs1 = 0.;
+    double c1 = 0.;
+    double cc1 = 0.;
+
+    double cs2 = 0.;
+    double ccs2 = 0.;
+    double c2 = 0.;
+    double cc2 = 0.;
+
+    for (auto input = first; input != last; input++) {
```

Comment:
Like above, we should create a variable `auto input_i_wgt = input->wgt;` at the head of the loop, and then we can minimize the dereferences.